backend/TraderX/src/main/java/cmpe451/group6/rest/portfolio/controller/PortfolioController.java

```diff
@@ -83,5 +83,14 @@ public PortfolioResponseDTO getPortfolio(HttpServletRequest req,
         return portfolioService.getPortfolio(util.unwrapUsername(req), portfolioName);
     }
   
+    @GetMapping(value = "/getSelfPortfolios")
+    @ResponseStatus(HttpStatus.OK)
+    @PreAuthorize("hasRole('ROLE_ADMIN') or hasRole('ROLE_BASIC') or hasRole('ROLE_TRADER')")
+    @ApiOperation(value = "Get all portfolios of requester", response = StringResponseWrapper.class)
+    @ApiResponses(value = { @ApiResponse(code = 400, message = GENERIC_ERROR_RESPONSE) })
+    public List<String> getSelfPortfolios(HttpServletRequest req) {
```

Comment:
I remember the return type of raw list caused problems when parsing the response on android side in the past. Creating a list wrapper class with the field of ‘List<Object>’ inside ‘helpers’ package might be good odea for both parsing and further use on backend side.